README.md

```diff
@@ -170,29 +167,60 @@ Arrays group a set of [values](#value-types), inside square brackets.
 key = []  // empty arrays are okay
 ```
 
+## Concatenation
+
+ZOMB files allow same-type value concatenation (string-string, object-object, and array-array) with the `+` operator. The following examples are boring, but they get the point across.
+
+```zomb
+key = bare_string + "quoted string" + \\raw-
+                                      \\string
+```
+
+```zomb
+key = { a = hello } + { b = world }
+```
+
+> _Keys in the resulting object must be unique._
+
+```zomb
+key = [ 1 2 3 ] + [ 4 5 6 ]
+```
+
+## Comments
+
+You've already seen comments in the previous examples, but now you know that comments are a real thing!
+
+Comments start with `//` and run to the end of the line.
+
+```zomb
+// this is a comment on its own line
+
+hello = goodbye // this is a comment at the end of a line
+
+key = [ // comments can be pretty much anywhere
+    1 2 3
+]
+```
+
 ## Macros
 
 Macros are the special sauce of ZOMB files. They allow you to write reusable values. Let's see an example:
 
 ```zomb
-$chuck = "Chuck Norris"
-$jobs = [ Hacker Dishwasher "Dog Walker" ]
-$names = {
-    god = Thor
-    file = ZOMB
-    "the best" = $chuck
+$color = {
+    black = #000000
+    red = #ff0000
+    hot_pink = #ff43a1
 }
 // sometimes commas are nice (still not required though)
-$person(name, job = "Software Engineer") = {
-    name = %name
-    job = %job
+$colorize(scope, color = $color.black) = {
+    scope = %scope
+    settings = { foreground = %color }
 }
-
-people = [
-    $person($names.file, $jobs.0)
-    $person($names.god, $jobs.1)
-    $person($names."the best", $jobs.2)
-    $person(Zooce)
+tokenColors = [
+    $colorize("editor.background", $color.black)
+    $colorize("editor.foreground", $color.red)
+    $colorize("comments", $color.hot_pink)
 ]
 ```
 
```

Comment:
I want to change how this is organized:
- Intro (above is fine)
- Defining a Macro
- Using a Macro (macro expressions)
- Macro Batching

Reply:
Done.